mirrord-cli/src/main.rs

```diff
@@ -60,19 +60,22 @@ const INJECTION_ENV_VAR: &str = "LD_PRELOAD";
 #[cfg(target_os = "macos")]
 const INJECTION_ENV_VAR: &str = "DYLD_INSERT_LIBRARIES";
 
-fn extract_library(dest_dir: Option<String>) -> String {
+fn extract_library(dest_dir: Option<String>) -> Result<String> {
     let library_file = env!("CARGO_CDYLIB_FILE_MIRRORD_LAYER");
     let library_path = std::path::Path::new(library_file);
     let file_name = library_path.components().last().unwrap();
     let file_path = match dest_dir {
         Some(dest_dir) => std::path::Path::new(&dest_dir).join(file_name),
         None => temp_dir().as_path().join(file_name),
     };
-    let mut file = File::create(file_path.clone()).unwrap();
+    let mut file = File::create(&file_path)
```

Comment:
We can't know for sure the reason it failed is due to path not existing, so I'd just add using context the file path ("Path .. creation failed")